### PR TITLE
Improves the OpInfo out= tests

### PR DESCRIPTION
**Pull request by mruberry**
Edit: OpInfos separated into their own PRs to debug an ASAN failure that doesn't identify the failing test properly. This PR now just updates the out tests.

Adds OpInfos for:

- nn.functional.smooth_l1_loss
- nn.functional.l1_loss
- nn.functional.pdist
- nn.functional.binary_cross_entropy
- nn.functional.triplet_margin_loss
- nn.functional.triplet_margin_with_distance_loss
- nn.functional.max_unpool{1, 2, 3}D
- nn.functional.alpha_dropout
- nn.functional.soft_margin_loss
- nn.functional.multilabel_soft_margin_loss
- nn.functional.multilabel_margin_loss
- nn.functional.multi_margin_loss
- nn.functional.margin_ranking_loss

These OpInfos were taken from https://github.com/pytorch/pytorch/pull/67560, https://github.com/pytorch/pytorch/pull/67823, https://github.com/pytorch/pytorch/pull/68625, and https://github.com/pytorch/pytorch/pull/67079. The sample input update from https://github.com/pytorch/pytorch/pull/67017 is also rolled into this PR.

cc @zou3519 @nikitaved @pmeier @vfdev-5 @dagitses

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -7654,22 +7658,88 @@ def sample_inputs_tensorsolve(op_info, device, dtype, requires_grad, **kwargs):
...
+    )
+
+    for shape in shapes:
+        for kwargs in [{}, {'margin': 1.0}]:
+            yield SampleInput(_make_tensor(shape),
+                              args=(_make_tensor(shape, requires_grad=False),
+                                    _make_tensor(shape, requires_grad=False)),
+                              kwargs=kwargs)
+
+# TODO: add reduction kwargs
```
Comment (pmeier) [inline torch/testing/_internal/common_methods_invocations.py:7715]: Why not use the `sample_inputs_loss` from above? That would handle the reduction.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -7654,22 +7658,88 @@ def sample_inputs_tensorsolve(op_info, device, dtype, requires_grad, **kwargs):
...
+        (S, S),
+    )
+
+    for shape in shapes:
+        # Produce one with weight and one without.
+        yield SampleInput(_make_tensor(shape), args=(_make_tensor(shape, requires_grad=False),), kwargs={})
+        yield SampleInput(_make_tensor(shape), args=(_make_tensor(shape, requires_grad=False),),
+                          kwargs={'weight': _make_tensor(shape, requires_grad=False)})
+
+# TODO: add reduction kwargs
```
Comment (pmeier) [inline torch/testing/_internal/common_methods_invocations.py:7730]: Same as above.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:7730]: It's not such an easy substitution. In this case the rhs has the long dtype, for example

**Comment by nikitaved** [comment]
@mruberry , https://github.com/pytorch/pytorch/pull/68625 not only adds OpInfos, it modifies some existing code for double backward which handles empty tensors.

**Comment by mruberry** [comment]
> @mruberry , #68625 not only adds OpInfos, it modifies some existing code for double backward which handles empty tensors.

Yep; but I'm only trying to add the OpInfos in this PR

**Comment by mruberry** [comment]
> Some samples that are being generated are empty, and the older code does not handle that.

Yep; I'm trying to add the appropriate skips and xfails.

**Pull request closed without merging**